Makefile

```diff
@@ -51,3 +53,19 @@ CONTROLLER_GEN=$(GOBIN)/controller-gen
 else
 CONTROLLER_GEN=$(shell which controller-gen)
 endif
+
+# Find or download setup-envtest
+setup-envtest:
+ifeq (, $(shell which setup-envtest))
+	@{ \
+	set -e ;\
+	SETUP_ENVTEST_TMP_DIR=$$(mktemp -d) ;\
+	cd $$SETUP_ENVTEST_TMP_DIR ;\
+	go mod init tmp ;\
+	go get sigs.k8s.io/controller-runtime/tools/setup-envtest@latest ;\
```

Comment:
Wouldn't it be more reliable to stick to an absolute version?
I think `go install` might be a better option now (since go 1.16) to manage installing these dev utils.

Reply:
Hi, thanks for reviewing.
I couldn't find any releases for setup-envtest in https://github.com/kubernetes-sigs/controller-runtime/releases and trying to install it with the overall controller-runtime release results in error

```console
go install sigs.k8s.io/controller-runtime/tools/setup-envtest@v0.9.2: module sigs.k8s.io/controller-runtime@v0.9.2 found, but does not contain package sigs.k8s.io/controller-runtime/tools/setup-envtest
```

I can pin it to a commit. Will that be okay?

Regarding the `go install` option, I followed what's done for controller-gen and gen-crd-api-reference-docs (in the source-controller repo), which is based on the kubebuilder generated makefile. `go install` works well. Should I update the controller-gen target to also use go install? and in other repos as well?

Reply:
I see that in the [readme](https://github.com/kubernetes-sigs/controller-runtime/tree/master/tools/setup-envtest#readme) they also reference `@latest` so maybe that's fine for now.

> Should I update the controller-gen target to also use go install? and in other repos as well?

I'm inclined to say yes, but I think it would make sense to check with the others as well. I think `go install` is more elegant, even though this [scaffolding](https://github.com/kubernetes-sigs/kubebuilder/blob/1831527c09c7ad54bd1782f0e01550b425ef9ae1/pkg/plugins/golang/v3/scaffolds/internal/templates/makefile.go#L60) is provided by kubebuilder by default. For now, you can probably leave it as is and we can explore the option to use `go install` later.